qemu/tests/blockdev_commit_hotunplug.py

```diff
@@ -0,0 +1,79 @@
+from virttest import env_process
+from virttest import utils_misc
+
+from provider import job_utils
+from provider import backup_utils
+from provider.blockdev_commit_base import BlockDevCommitTest
+
+
+class BlockdevCommitHotunplug(BlockDevCommitTest):
+
+    def is_device_deleted(self, device):
+        return device not in str(self.main_vm.monitor.info_block())
+
+    def commit_snapshots(self):
+        device = self.params.get("device_tag")
+        device_params = self.params.object_params(device)
+        snapshot_tags = device_params["snapshot_tags"].split()
+        self.device_node = self.get_node_name(device)
+        options = ["speed"]
+        arguments = self.params.copy_from_keys(options)
+        arguments["speed"] = self.params["commit_speed"]
+        device = self.get_node_name(snapshot_tags[-1])
+        commit_cmd = backup_utils.block_commit_qmp_cmd
+        cmd, args = commit_cmd(device, **arguments)
+        self.main_vm.monitor.cmd(cmd, args)
+        self.main_vm.monitor.cmd('device_del',
+                                 {'id': self.params["device_tag"]})
+        unplug_s = utils_misc.wait_for(lambda: self.is_device_deleted(device),
+                                       timeout=60, step=1.0)
+        if not unplug_s:
+            self.test.fail("Hotunplug device failed")
+        job_status = str(job_utils.query_block_jobs(self.main_vm))
+        if self.params["expect_status"] not in job_status:
+            self.test.fail("Job status not correct,job status is %s"
+                           % job_status)
+        job_id = args.get("job-id", device)
+        self.main_vm.monitor.cmd("block-job-set-speed",
+                                 {'device': job_id, 'speed': 0})
+        job_utils.wait_until_block_job_completed(self.main_vm, job_id)
+
+    def vm_with_commited_disk(self):
+        """Boot VM with commited data disks"""
+        if self.main_vm.is_alive():
+            self.main_vm.destroy()
+
+        vm_name = self.params["main_vm"]
+        vm_params = self.params.object_params(vm_name)
+        system_image = self.params.objects("images")[0]
+        images = [system_image]
+        images += [self.params["snapshot_tags_data"]]
```

Comment:
In my opinion, after commit, data from top device(sn1) has been written to base(data), so we need to check files in data device instead of sn1, @leidwang @ngu-niny please correct me if anything wrong, thanks.
Then if I have a good understanding on commit, here, you just need to vm.destroy(), then, vm.create(), vm.verify_alive()

Reply:
@leidwang @ngu-niny , I just checked the current automated commit test cases, vm NEVER restarted, we just verify_data_file() right after commit_snapshots(), so here please make a decision if you'd like to keep the current test steps, 
if yes, @leidwang , don't need to restart VM, just wait commit job done, run_test in base class has already called verify_data_file to do file verification;
if no, I am afraid we have to modify blockdev_commit_base module, to update the testing process:
commit->restart vm with base image->mount image and do file verification.

Either is OK for me.

Reply:
I think we can keep the current test steps, just wait commit job done.And I will sync the status with @aliang123 when she comes back.

Reply:
@leidwang @zhencliu If so, I prefer to use the existing commit test, verifying the data is enough, while restart vm is not a must. Thanks for the endeavour @leidwang.

Reply:
Thank you for your comments, then @leidwang , we can remove the restart vm function.

Reply:
I have updated the code,could you please help review this patch again?Thanks a lot.